docs/references/beta-only-features.md

```diff
@@ -0,0 +1,89 @@
+# Adding features to be released in the beta channel (only)
+
+The primary NPM package released from this repository is @azure/communication-react.
+
+We maintain two "release channels" for this package.
+
+* Stable releases - follow semantic versioning, e.g. 1.0.0
+* Beta releases - do not follow sematnic versioning. These are packages with a `-beta.X` suffix, e.g. 1.0.1-beta.3
+
+Our beta releases often carry features that are not yet stabilized. A feature may be restricted to beta channel for
+one of two reasons.
+
+* A feature that is actively being developed. Fature will be released in a stable package once it is beta-tested.
+* A feature that depends on a feature from underlying Azure Communication Service features that are in themselves in the beta phase.
+  This feature can only be released in a stable package once the underlying features are released in a stable package.
+
+The key difference from many standard NPM packages is that some features can stay in the beta packages across intervening stable package releases.
+
+# Conditional compilation
+
+We follow a green-trunk development practice. This means that all development happens on the `main` branch, and all releases are generated by creating
+short-lived branches off of `main`. The need to carry some features that are released in the beta channel but not in intervening stable package releases means
+that we need a way to enable / disable these features in the releases. Additionally, we need to be able to switch between beta and stable dependencies for
+underlying Azure Communication Service libraries.
+
+We achieve this by conditionally compiling away code to drop features from the stable package release.
+
+All build, test, local development server scripts can compile the code in this repository in one of two flavors:
+
+* `beta`: This is the default build flavor. It includes all code in the packges, and depends on the beta flavors of some of our dependencies.
+* `stable`: Conditionally compiled code is removed when building this flavor. It also depends on only the stable flavors of our dependencies.
+
+# Tooling
+
+To switch between flavors:
+
+```bash
+rush switch-flavor:stable
+# Need to rebuild from scratch because that command cleans all build caches.
+rush build
+```
+
+To switch back to beta flavor:
+
+```bash
```

Comment:
Maybe we could also add a set of command, telling people how to navigate into one project and only build that one when they are encountering some building issues of stable build? rush build globally sometimes doesn't do a good job to report build errors